R/init.R

```diff
@@ -122,52 +119,57 @@ init <- function(data,
       )
     )
   }
-
+  if (is.list(data) && !inherits(data, "teal_data_module")) {
+    checkmate::assert_list(data, names = "named")
+    data <- do.call(teal.data::teal_data, data)
+  }
   checkmate::assert_multi_class(data, c("teal_data", "teal_data_module"))
-  checkmate::assert_multi_class(modules, c("teal_module", "list", "teal_modules"))
+
```

Comment:
```suggestion
  checkmate::assert(
    checkmate::assert_multi_class(modules, c("teal_module", "teal_modules")),
    checkmate::assert_list(modules, c("teal_module", "teal_modules"))
  )
```

Same to @chlebowa and @averissimo comments below I prefer assert and then convert. I Also prefer assert error to have informative message. I admit that previously `assert_multi_class` wasn't precise enough.

Reply:
Applied assert and convert. Please see the new message in the main PR theread. Also converted few assert_multi_class into a assert. If you spot anything more - please kindly put a new comment